pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -178,7 +178,7 @@ func getAzurePipelineRequest(ctx context.Context, url string, metadata *azurePip
 		return []byte{}, err
 	}
 
-	req.SetBasicAuth("PAT", metadata.personalAccessToken)
+	req.SetBasicAuth("", strings.TrimSuffix(metadata.personalAccessToken, "\n"))
```

Comment:
I'm not sure if removing the header PAT it'll work (let's see the e2e test how it finishes). I agree with removing any new line at the end, but please do it in the metadata parsing to do it only once.
Could you add a unit test for this please?

Reply:
@JorTurFer is there already a unit test? https://github.com/kedacore/keda/blob/main/tests/scalers/azure-pipelines.test.ts. Is there something more specific that I can add to that test?

Reply:
hey,
yes, there is a file for [unit tests](https://github.com/kedacore/keda/blob/main/pkg/scalers/azure_pipelines_scaler_test.go). You could add this check to the `TestParseAzurePipelinesMetadata` as a validation in [the api stub](https://github.com/kedacore/keda/blob/main/pkg/scalers/azure_pipelines_scaler_test.go#L42).

Also remove this line because it's duplicated (at least in GH UI :/ ) 
![image](https://user-images.githubusercontent.com/36899226/174112017-74087269-fea6-444f-9c93-34cd7cd082fa.png)